lib/matplotlib/image.py

```diff
@@ -966,7 +970,21 @@ def set_extent(self, extent):
         state is not changed, so following this with ``ax.autoscale_view()``
         will redo the autoscaling in accord with ``dataLim``.
         """
-        self._extent = xmin, xmax, ymin, ymax = extent
+        (xmin, xmax), = self.axes._process_unit_info(
+            [("x", [extent[0], extent[1]])], kwargs)
+        (ymin, ymax), = self.axes._process_unit_info(
+            [("y", [extent[2], extent[3]])], kwargs)
```

Comment:
You can do both together, I think?
```suggestion
        (xmin, xmax), (ymin, ymax) = self.axes._process_unit_info(
            [("x", [extent[0], extent[1]]),
             ("y", [extent[2], extent[3]])],
            kwargs)
```